src/react/ui-elements/__tests__/SelectAccountIAMRole.test.tsx

```diff
@@ -0,0 +1,372 @@
+              <Description>Has all permissions and full access to the 100 account resources and manages ARTESCA users.</Description>
+              <Path>/scality-internal/</Path>
+              <RoleName>storage-manager-role</RoleName>
+              <RoleId>YRA3NTDUTWN6DRN76LSSDM6HA22RWBO9</RoleId>
+              <Arn>arn:aws:iam::232853836441:role/scality-internal/storage-manager-role</Arn>
+              <CreateDate>2024-04-17T16:31:36Z</CreateDate>
+            </member>
+          </Roles>
+          <IsTruncated>false</IsTruncated>
+        </ListRolesResult>
+        <ResponseMetadata>
+          <RequestId>148012f42345b8eb7c29</RequestId>
+        </ResponseMetadata>
+      </ListRolesResponse>
+        `),
+      );
+    }
+
+    if (params.get('Action') === 'GetRolesForWebIdentity') {
+      const TEST_ACCOUNT =
+        USERS.find((user) => user.id === testAccountId)?.userName ?? '';
+      const TEST_ACCOUNT_CREATION_DATE =
+        USERS.find((user) => user.id === testAccountId)?.createDate ?? '';
+
+      return res(
+        ctx.json({
+          IsTruncated: false,
+          Accounts: [
+            {
+              Name: TEST_ACCOUNT,
+              CreationDate: TEST_ACCOUNT_CREATION_DATE,
+              Roles: [
+                {
+                  Name: 'storage-manager-role',
+                  Arn: `arn:aws:iam::${testAccountId}:role/scality-internal/storage-manager-role`,
+                },
+              ],
+            },
+          ],
+        }),
+      );
+    }
+  });
+};
+
+const server = setupServer(getConfigOverlay(TEST_API_BASE_URL, INSTANCE_ID));
+
+const LocalWrapper = ({ children }) => {
+  const queryClient = new QueryClient({
+    // In test environnement, we don't want to retry queries
+    // because we may test the error case
+    defaultOptions: {
+      queries: {
+        retry: false,
+      },
+      mutations: {
+        retry: false,
+      },
+    },
+  });
+
+  return (
+    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
+  );
+};
+
+describe('SelectAccountIAMRole', () => {
+  const seletors = {
+    accountSelect: () => screen.getByText(/select account/i),
+    roleSelect: () => screen.getByText(/select role/i),
+    selectOption: (name: string | RegExp) =>
+      screen.getByRole('option', {
+        name: new RegExp(name, 'i'),
+      }),
+    loadingAccount: () => screen.getByText(/Loading.../i),
+  };
+  beforeAll(() => {
+    server.listen({ onUnhandledRequest: 'error' });
+  });
+  afterEach(() => {
+    server.resetHandlers();
+  });
+  afterAll(() => {
+    server.close();
+  });
+  it('renders with normal props', async () => {
```

Comment:
I think you want an additional test where you change account and change roles for this account and check if the options of the role select matches